### DISCUSSION TITLE
Expose GPLv3 as top level license for ansible-lint

### DISCUSSION THREAD

**Discussion by ssbarnea** [ansible/ansible-lint#1832] [Ideas]
While the change from https://github.com/ansible-community/ansible-lint/pull/1213 helped a little bit with GPLv3  aspects of ansible-lint, I think it needs few extra changes, especially because:

* removal of ansible-core (GPLv3) as a dependency is not possible in the foreseeable future
* we want to make `yamllint`  (GPLv3) a permanent dependency, so the tool will not have different outcomes based on when it is installed or not
* presence of `MIT License` on github license or on pypi may confuse some users, making them imagine that they can really use ansible-lint without being affected by GPLv3 implications.
* `NoLogPasswordsRule.py` is Apache (by Rackspace), which is not the same a MIT, but very close.

Thus I propose the below changes:
* List only `GPLv3` in package metadata
* Rename current MIT license from `LICENSE` to `LICENSE.MIT` and add `GPLv3` license to `LICENSE` file, as that is the default file where github looks for license. The outcome is that github replace the potentially misleading hyperlink name from `MIT License` to `License`.

While I (not a lawyer) do think that licensing the project a GPLv3 only is also possible, I do not see a need to do this and the `README.md` note about contributions as being under MIT is not really a problem.

I do hope that this move to address the few concerns I seen over the last years where one or two individuals claimed that ansible-lint is in violation of GPLv3 license due to the imports from ansible projects.  If we make these changes we can make them happy and likely also allow us to make all runtime dependencies permanent instead of listing them as optional, making the tool behavior more predictable.

What do you think about that?

[the remark this answers is not in the document]
**Reply by gundalow** [reply]
Please contact Red Hat legal, I've sent details internally.

[the remark this answers is not in the document]
**Reply by ssbarnea** [reply]
We already did that, and is mentioned in #1188, ticket id `LOS0549210`, waiting for their reply.

**Comment by sivel** [comment]
Just because I contacted the FSF about this a while back, I'll provide the reply I received from them:

> From what you've described I agree with your conclusion. While the parts
> of ansible-lint which are under GPL-compatible free software licenses
> such as the Expat (aka MIT) license may remain under that license, the
> work as a whole must be distributed under the terms of the GPL when
> incorporating GPL'd works.

My "conclusion" referenced in the reply was:

> Based on my understanding, this would make ansible-lint a derivative work of ansible, and should therefore carry the GPLv3 license.
